lib/main.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:js_interop';
+import 'dart:js_util';
+
+import 'package:path/path.dart' as path;
+import 'package:setup_dart/node/actions/tool_cache.dart';
+
+import 'node/actions/core.dart';
+import 'node/actions/exec.dart';
+import 'node/actions/http_client.dart';
+import 'node/fs.dart';
+import 'node/os.dart';
+import 'node/process.dart';
+
+void main(List<String> args) async {
+  try {
+    // sdk
+    var sdk = core.getInput('sdk');
+    if (sdk.isEmpty) {
+      sdk = 'stable';
+    }
+
+    // flavor
+    var flavor = core.getInput('flavor');
+    if (flavor.isEmpty) {
+      flavor = sdk == 'main' ? 'raw' : 'release';
+    } else if (flavor != 'raw' && flavor != 'release') {
+      core.setFailed("Unrecognized build flavor '$flavor'.");
+      return;
+    }
+    final raw = flavor == 'raw';
+
+    // os
+    final os = getPlatform();
+
+    // architecture
+    var architecture = core.getInput('architecture');
+    if (architecture.isEmpty) {
+      architecture = getArch();
+    }
+
+    // calculate version and channel
+    String version;
+    String channel;
+
+    if (sdk == 'stable' || sdk == 'beta' || sdk == 'dev') {
+      channel = sdk;
+      version =
+          raw ? 'latest' : (await latestPublishedVersion(channel, flavor))!;
+    } else if (sdk == 'main') {
+      channel = 'be';
+      version =
+          raw ? 'latest' : (await latestPublishedVersion(channel, flavor))!;
+    } else {
+      version = sdk;
+
+      // Derive the channel from the version string.
+      if (sdk.contains('dev')) {
+        channel = 'dev';
+      } else if (sdk.contains('beta')) {
+        channel = 'beta';
+      } else if (sdk.contains('main')) {
+        core.setFailed('Versions cannot be specified for main channel builds.');
+        return;
+      } else {
+        channel = 'stable';
+      }
+    }
+
+    core.info('Installing the $os-$architecture Dart SDK version $version from '
+        'the $channel ($flavor) channel.');
+
+    // Calculate url based on https://dart.dev/tools/sdk/archive#download-urls.
+    final url = 'https://storage.googleapis.com/dart-archive/'
+        'channels/$channel/$flavor/$version/sdk/'
+        'dartsdk-$os-$architecture-release.zip';
+
+    // Use a cached sdk or download and cache the sdk; using a 'raw' sdk flavor
+    // disables caching.
+    final toolName = flavor == 'raw' ? 'dart_raw' : 'dart';
```

Comment:
Replace `flavor == 'raw'` with `raw` like in the next line.

Reply:
done